test/view/all.cpp

```diff
@@ -22,21 +22,27 @@ int main()
 
     int rgi[] = {1, 1, 1, 2, 3, 4, 4};
     std::vector<int> vi(begin(rgi), end(rgi));
-    std::list<int> li(begin(rgi), end(rgi));
+    std::list<int> const li(begin(rgi), end(rgi));
 
-    iterator_range<int *> x = view::all(rgi);
-    iterator_range<std::vector<int>::iterator> y = view::all(vi);
-    sized_iterator_range<std::list<int>::iterator> z = view::all(li);
+    auto x = view::all(rgi);
+    ::has_type<ref_view<decltype(rgi)> &>(x);
+    CONCEPT_ASSERT(range_cardinality<decltype(x)>::value == 7);
+    auto y = view::all(vi);
+    ::has_type<ref_view<decltype(vi)> &>(y);
+    auto z = view::all(li);
+    ::has_type<ref_view<decltype(li)> &>(z);
 
     x = view::all(x);
     y = view::all(y);
     z = view::all(z);
 
-    CHECK(x.size() == 7u);
-    CHECK(y.size() == 7u);
-    CHECK(z.size() == 7u);
+    CHECK(ranges::size(x) == 7u);
+    CHECK(ranges::size(y) == 7u);
+    CHECK(ranges::size(z) == 7u);
```

Comment:
This change isn't *incorrect* but it's unnecessary: `ref_view<SizedRange>` always has member `size`. Let's put this back.